flink-end-to-end-tests/flink-tpcds-test/src/main/java/org/apache/flink/table/tpcds/utils/TpcdsResultComparator.java

```diff
@@ -0,0 +1,232 @@
+			BufferedReader actualReader = new BufferedReader(new FileReader(actualFile));
+
+			int expectedLineNum = 0;
+			int actualLineNum = 0;
+
+			String expectedLine, actualLine;
+			while ((expectedLine = expectedReader.readLine()) != null &&
+				(actualLine = actualReader.readLine()) != null) {
+				expectedLineNum++;
+				actualLineNum++;
+
+				// reslut top 8 line of query 34,
+				// result line  2、3  0f query 77
+				// result line 18、 19 of query 79
+				// have different order with answer set, because of Flink keep nulls last for DESC, nulls first for ASC.
+				// it's up to TPC-DS standard.
+				if ("34.ans".equals(expectedFile.getName()) && expectedLineNum == 1) {
+					List<String> expectedTop8Line = new ArrayList<>(8);
+					List<String> actualTop8Line = new ArrayList<>(8);
+
+					String expectedLine2 = expectedReader.readLine();
+					expectedLineNum++;
+					while (expectedLineNum < 8) {
+						expectedTop8Line.add(expectedReader.readLine());
+						expectedLineNum++;
+					}
+					expectedTop8Line.add(expectedLine);
+					expectedTop8Line.add(expectedLine2);
+
+					actualTop8Line.add(actualLine);
+					while (actualLineNum < 8) {
+						actualTop8Line.add(actualReader.readLine());
+						actualLineNum++;
+					}
+					for (int i = 0; i < 8; i++){
+						if (!isEqualLine(expectedTop8Line.get(i), actualTop8Line.get(i), i)) {
+							return false;
+						}
+					}
+					continue;
+				}
+				if ("77.ans".equals(expectedFile.getName()) && expectedLineNum == 2) {
+					String expectedLine3 = expectedReader.readLine();
+					String actualLine3 = actualReader.readLine();
+					expectedLineNum++;
+					actualLineNum++;
+
+					if (isEqualLine(expectedLine, actualLine, 2) && isEqualLine(expectedLine3, actualLine3, 3)) {
+						continue;
+					}
+					if (isEqualLine(expectedLine, actualLine3, 2) && isEqualLine(expectedLine3, actualLine, 3)) {
+						continue;
+					}
+					return false;
+				}
+
+				if ("79.ans".equals(expectedFile.getName()) && expectedLineNum == 18) {
+					String expectedLine19 = expectedReader.readLine();
+					String actualLine19 = actualReader.readLine();
+					expectedLineNum++;
+					actualLineNum++;
+
+					if (isEqualLine(expectedLine, actualLine, 18) && isEqualLine(expectedLine19, actualLine19, 19)) {
+						continue;
+					}
+					if (isEqualLine(expectedLine, actualLine19, 18) && isEqualLine(expectedLine19, actualLine, 19)) {
+						continue;
+					}
+					return false;
+				}
+
+				if (!isEqualLine(expectedLine, actualLine, expectedLineNum)) {
+					return false;
+				}
+			}
+
+			while (expectedReader.readLine() != null) {
+				expectedLineNum++;
+			}
+			while (actualReader.readLine() != null) {
+				actualLineNum++;
+			}
+
+			if (expectedLineNum != actualLineNum) {
+				System.out.println(
+					"[ERROR]incorrect number of lines! Expecting " + expectedLineNum +
+						" lines, but found " + actualLineNum + " lines.");
+				return false;
+			}
+		} catch (Exception e) {
+			e.printStackTrace();
```

Comment:
Is it possible to let the exceptions bubble up and simply let the process exit with the stack trace?